src/ensembl/src/shared/components/view-in-app/ViewInApp.tsx

```diff
@@ -45,7 +45,7 @@ export type urlObj = Record<AppName, string>;
 
 export type ViewInAppProps = {
   links: Partial<urlObj>;
-  push: Push;
+  shouldReplaceState?: boolean;
```

Comment:
This looks far too indiscriminate. Different links may have different behaviour. I think this should go into the urlObj; and the type of the urlObj should be updated to:

```
{
  [appName: AppName]: {
    link: string;
    replaceState: boolean
  }
}
```

Reply:
That's done now. Did some small changes to `urlObj` too.

Reply:
✅

Reply:
>  Did some small changes to urlObj too.

Hmm, interesting. I was expecting the urlObj to always have the values of `{ link: string, replaceState?: boolean}`, but your suggestion may be more convenient. We'll discuss this during the code review.